server/server.js

```diff
@@ -1,83 +1,95 @@
+const mainRoutes = require("./routes/main");
+
 
 // Passport config
 require("./config/passport")(passport);
 
-//Body Parsing
-app.use(express.urlencoded({ extended: true }));
-app.use(express.json());
+(async () => {
+  //Body Parsing
+  app.use(express.urlencoded({ extended: true }));
+  app.use(express.json());
 
-//Logging
-app.use(logger("dev"));
+  //Logging
+  app.use(logger("dev"));
 
 
-// Setup Sessions - stored in MongoDB
-app.use(
-  session({
-    secret: process.env.SESSION_SECRET || "keyboard cat",
-    resave: false,
-    saveUninitialized: false,
-    store: new MongoStore({ mongooseConnection: mongoose.connection }),
-  })
-);
+  // Setup Sessions - stored in MongoDB
+  app.use(
+    session({
+      secret: process.env.SESSION_SECRET || "keyboard cat",
+      resave: false,
+      saveUninitialized: false,
+      store: new MongoStore({ mongooseConnection: mongoose.connection }),
+    })
+  );
+
+  // Render React as View
+  // app.use(express.static(path.join(__dirname, "..", "client", "build")));
+
+  // Passport middleware
+  app.use(passport.initialize());
+  app.use(passport.session());
 
-// Render React as View
-app.use(express.static(path.join(__dirname, "..", "client", "build")));
+  if (process.env.NODE_ENV === "development" && process.env.MOCK_USER === "true") {
+    console.log("In development - using mocked user")
+    app.use((req, res, next) => {
+      req.user = mockUser;
+      next()
+    })
+  }
 
-// Passport middleware
-app.use(passport.initialize());
-app.use(passport.session());
+  //Use flash messages for errors, info, ect...
+  app.use(flash());
 
-if (process.env.NODE_ENV === 'development' && process.env.MOCK_USER === 'true') {
-  console.log("In development - using mocked user")
-  app.use((req, res, next) => {
-    req.user = mockUser;
-    next()
+  //Setup Routes For Which The Server Is Listening
+  app.use("/", mainRoutes);
+  app.use("/events", eventsRoutes);
+
+  // 404 handler
+  // No idea why this was causing 404 error for all routes
+  // app.use((req, res) => {
+  //   res.status(404).json({ message: "Not found" });
+  // });
+
+  // error handler
+  app.use((err, req, res, next) => {
+    const { status = 500, message = "Server error", stack } = err;
+    console.log(stack);
+    res.status(status).json({ message });
+  });
+
+  const nextApp = await prepareNextApp();
+  const nextHandler = nextApp.getRequestHandler();
+  const Database = await connectDB();
+
+  // Set Next Routes
+  app.get("*", (req, res) => {
+    nextHandler(req, res, parse(req.url, true));
   })
-}
-
-//Use flash messages for errors, info, ect...
-app.use(flash());
-
-//Setup Routes For Which The Server Is Listening
-app.use("/", mainRoutes);
-app.use("/events", eventsRoutes);
-app.get("'", (req, res) => {
-  res.sendFile(path.join(__dirname, "..", "client", "build", "index.html"));
-});
-
-// 404 handler
-app.use((req, res) => {
-  res.status(404).json({ message: "Not found" });
-});
```

Comment:
No idea why this was causing NextJs to serve a 404 with json rather than the correct page component.